crates/runtime/src/search/rrf.rs

```diff
@@ -484,113 +671,203 @@ mod tests {
-        }))
+        register_udfs(&rt);
+        Ok(rt)
     }
 
-    async fn make_test_runtime() -> Result<Runtime> {
-        let rt = RuntimeBuilder::new().build().await;
-        rt.df
-            .ctx
-            .state()
-            .config_mut()
-            .set_extension(Arc::clone(&TEST_REQUEST_CONTEXT));
+    async fn make_fruit_dataframe(runtime: &Runtime) -> Result<DataFrame> {
+        let rowid_expr = row_number()
+            .order_by(vec![col("content").sort(false, false)])
+            .build()?
+            .alias("id");
 
-        let test_table = make_test_table(&[
-            "banana yellow curved fruit",
-            "orange citrus round juicy",
-            "apple fruit sweet red crispy",
-        ])?;
-        rt.df
+        let df = runtime
+            .df
             .ctx
-            .register_table("foo", test_table)
-            .expect("Failed to register foo table");
+            .sql(
+                "SELECT
+              unnest([
+                  'banana yellow curved fruit',
+                  'orange citrus round juicy',
+                  'apple fruit sweet red crispy'
+              ]) as content",
+            )
+            .await?;
 
-        register_udfs(&rt);
-        Ok(rt)
+        let embed_expr = df.parse_sql_expr("embed(content, 'test_model')")?;
+
+        df.window(vec![rowid_expr])?
+            .with_column("content_embedding", embed_expr)
+    }
+
+    fn stub_scalar_function(name: &str) -> Expr {
+        let stub_udf = create_udf(
+            name,
+            vec![DataType::Utf8; 0],
+            DataType::Utf8,
+            Volatility::Stable,
+            Arc::new(|_| {
+                Ok(ColumnarValue::Scalar(ScalarValue::Utf8(Some(
+                    "stub".to_string(),
+                ))))
+            }),
+        );
+
+        Expr::ScalarFunction(ScalarFunction::new_udf(Arc::new(stub_udf), vec![]))
     }
 
     #[tokio::test(flavor = "multi_thread")]
-    async fn test_fuse_queries() {
-        let runtime = make_test_runtime()
-            .await
-            .expect("Failed to create test runtime");
+    async fn test_recency_scoring() -> Result<ExitCode> {
+        let runtime = make_test_runtime().await?;
+
+        let fruit_df = make_fruit_dataframe(&runtime)
+            .await?
+            .with_column("picked_at", now())?
+            .with_column(
+                "picked_at",
+                to_unixtime(vec![col("picked_at")]) - (lit(43200) * col("id")),
+            )?;
+
+        let picked_at_expr = fruit_df.parse_sql_expr("to_timestamp(cast(picked_at as bigint))")?;
 
-        let query =
-            "select * from rrf(vector_search(foo, 'crispy'), vector_search(foo, 'red'), id, 600.0)";
+        let fruit_df = fruit_df
+            .with_column("picked_at", picked_at_expr)?
+            .sort(vec![col("picked_at").sort(false, false)])?;
+
+        let fruit_embedding_table = df_as_embedding_table(&runtime, fruit_df.clone())?;
```

Comment:
Thanks @kczimm for showing me `DataFrame::into_view`. This is something that I missed a lot from writing tests in Spark. It is _so much easier_ to set up test data with a DataFrame (imagine all of the record batch building + coercing to do the equivalent of what we do to build `fruit_df`).